### PR TITLE
fix(mqtt): new address monitoring

### PR DESCRIPTION
**Pull request by lucasfernog**
# Description of change

Properly `.await` on mqtt async tasks.

## Links to any relevant issues

N/A

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

CLI wallet, getting funds on new address and checking if MQTT is subscribed/responding.

### CODE REVIEW

File: src/monitor.rs [hunk 1]
```diff
@@ -145,7 +145,7 @@ async fn process_output(
     let client_options_ = client_options.clone();
 
     if address_wrapper == latest_address {
-        account_handle.generate_address_internal(&mut account).await?;
+        let _ = account_handle.generate_address_internal(&mut account).await;
```
Comment (melatron) [inline src/monitor.rs:150]: Why is the error handling removed? Maybe a comment would be nice.

File: src/account/sync/mod.rs [hunk 2]
```diff
@@ -1589,7 +1589,7 @@ async fn perform_transfer(
 
     for address in addresses_to_watch {
         // ignore errors because we fallback to the polling system
-        let _ = crate::monitor::monitor_address_balance(account_handle.clone(), &address);
+        let _ = crate::monitor::monitor_address_balance(account_handle.clone(), &address).await;
```
Comment (melatron) [inline src/account/sync/mod.rs:1592]: Why is `let _ =` needed? It is not returning a Future any more.

File: src/account/mod.rs [hunk 3]
```diff
@@ -522,11 +522,19 @@ impl AccountHandle {
...
+        // monitor on a non-async function to prevent cycle computing the `monitor_address_balance` fn type
+        self.monitor_address(address.address().clone());
 
         Ok(address)
     }
 
+    pub(crate) fn monitor_address(&self, address: AddressWrapper) {
+        let handle = self.clone();
+        crate::spawn(async move {
+            let _ = crate::monitor::monitor_address_balance(handle, &address).await;
```
Comment (melatron) [inline src/account/mod.rs:535]: Why is `let _` = needed?

File: src/account/sync/mod.rs (hunk 2 above)
Comment (lucasfernog) [reply src/account/sync/mod.rs:1592]: this one has a comment in there :P

File: src/monitor.rs (hunk 1 above)
Comment (lucasfernog) [reply src/monitor.rs:150]: you're right, done

File: src/account/mod.rs (hunk 3 above)
Comment (lucasfernog) [reply src/account/mod.rs:535]: added a comment

**Review by melatron — APPROVED** [review]

**Pull request merged**